libsrc/apple2/ser/a2gs.ssc.s

```diff
@@ -0,0 +1,633 @@
+;
+; Serial driver for the Apple IIgs Zilog Z8530.
+;
+; Colin Leroy-Mira <colin@colino.net>, 2023
+;
+; This software is licensed under the same license as cc65,
+; the zlib license (see LICENSE file).
+;
+; Documentation from http://www.applelogic.org/files/Z8530UM.pdf (pages
+; referred to where applicable)
+; and https://gswv.apple2.org.za/a2zine/Utils/Z8530_SCCsamples_info.txt
+
+
+
+.ifdef  __APPLE2ENH__
+        .setcpu         "65816"
```

Comment:
Do we really need that?

Reply:
Sadly yes, for this: https://archive.org/details/IIgs_2523018_SCC_Access/page/n7/mode/2up
We have to inform the ROM that we want serial IRQs sent to our IRQ handler, using a flag at $E10104 - which of course is an illegal address for 65(c)02.

Can you think of a workaround for that?

Reply:
Sorry, I didn't want to imply that this is a problem. I was just wondering if it is necessary as I didn't see any code transitioning from emulation mode to native mode. I'm not an experienced 65816 programmer, so I wasn't aware that 24-bit addresses are available in emulation mode as well. Now I understand and everything is fine :-)